src/main/java/org/tron/keystore/WalletFile.java

```diff
@@ -0,0 +1,461 @@
+                ? !getCrypto().equals(that.getCrypto())
+                : that.getCrypto() != null) {
+            return false;
+        } 
+        if (getId() != null 
+                ? !getId().equals(that.getId())
+                : that.getId() != null) {
+            return false;
+        }
+        return version == that.version;
+    } 
+
+    @Override
+    public int hashCode() {
+        int result = getAddress() != null ? getAddress().hashCode() : 0;
+        result = 31 * result + (getCrypto() != null ? getCrypto().hashCode() : 0);
+        result = 31 * result + (getId() != null ? getId().hashCode() : 0);
+        result = 31 * result + version;
+        return result;
+    }  
+
+    public static class Crypto {
+        private String cipher;
+        private String ciphertext;
+        private CipherParams cipherparams;
+
+        private String kdf;
+        private KdfParams kdfparams;
+
+        private String mac;
+
+        public Crypto() {
+        }
+
+        public String getCipher() {
+            return cipher;
+        }
+
+        public void setCipher(String cipher) {
+            this.cipher = cipher;
+        }
+
+        public String getCiphertext() {
+            return ciphertext;
+        }
+
+        public void setCiphertext(String ciphertext) {
+            this.ciphertext = ciphertext;
+        }
+
+        public CipherParams getCipherparams() {
+            return cipherparams;
+        }
+
+        public void setCipherparams(CipherParams cipherparams) {
+            this.cipherparams = cipherparams;
+        }
+
+        public String getKdf() {
+            return kdf;
+        }
+
+        public void setKdf(String kdf) {
+            this.kdf = kdf;
+        }
+
+        public KdfParams getKdfparams() {
+            return kdfparams;
+        }
+
+        @JsonTypeInfo(
+                use = JsonTypeInfo.Id.NAME,
+                include = JsonTypeInfo.As.EXTERNAL_PROPERTY,
+                property = "kdf")
+        @JsonSubTypes({
+                @JsonSubTypes.Type(value = Aes128CtrKdfParams.class, name = Wallet.AES_128_CTR),
+                @JsonSubTypes.Type(value = ScryptKdfParams.class, name = Wallet.SCRYPT)
+        })
+        // To support my Ether Wallet keys uncomment this annotation & comment out the above
+        //  @JsonDeserialize(using = KdfParamsDeserialiser.class)
+        // Also add the following to the ObjectMapperFactory
+        // objectMapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
+        public void setKdfparams(KdfParams kdfparams) {
+            this.kdfparams = kdfparams;
+        }
+
+        public String getMac() {
+            return mac;
+        }
+
+        public void setMac(String mac) {
+            this.mac = mac;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o.getClass() != Crypto.class) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Compare to "this.getClass()" instead. ([squid:S2162](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2162))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=938&issues=AWPK23iWkDisRwWBfuxH&open=AWPK23iWkDisRwWBfuxH)

[ISSUE_KEY]: AWPK23iWkDisRwWBfuxH